application_form/services/reservation.py

```diff
@@ -88,11 +109,33 @@ def create_reservation_without_application(
         if user:
             reservation_data["handler"] = user.profile_or_user_full_name
 
+        ownership_type = apartment.project_ownership_type
+        right_of_residence_ordering_number = reservation_data[
+            "customer"
+        ].right_of_residence_ordering_number
+
+        new_list_position = (max_list_position or 0) + 1
+        new_queue_position = (max_queue_position or 0) + 1
+
+        if ownership_type.lower() == "haso":
+            late_reservations = existing_reservations.filter(submitted_late=True)
+            if late_reservations:
+                new_queue_position = calculate_new_queue_position(
+                    late_reservations,
+                    right_of_residence_ordering_number,
+                    max_queue_position,
+                )
+                new_list_position = new_queue_position
+                _make_room_for_reservation(
+                    apartment_uuid, new_list_position, new_queue_position
+                )
+
         reservation = ApartmentReservation(
             **reservation_data,
             state=state,
-            list_position=(max_list_position or 0) + 1,
-            queue_position=(max_queue_position or 0) + 1,
+            list_position=new_list_position,
+            submitted_late=True,
```

Comment:
fixed, submitted_late is set because this is called only via Add Application from the sales-ui admin, which appears only after the lottery has been run